src/Models.js

```diff
@@ -1,47 +1,42 @@
-function getExp(skillName){
-  /* Gets exp of a skill from storage.
+function getLevelAndLevelUpExp(skillNr) {
+  /* Gets exp from storage and Calculates current level and exp
+   * needed to reach next level.
+   * O(max_level) - computational complexity
    */
   return new Promise((resolve, reject) => {
+    var skillName = allSkills[skillNr];
     chrome.storage.sync.get([skillName], function (result) {
-      var overallExp;
-      if (skillName in result) {
-        overallExp = result[skillName];
+      var exp = 0;
+      if(skillName in result) {
+        exp = result[skillName];
       }
-      else{
-        overallExp = 0;
+      var level = 0;
+      while(exp >= expTable[level + 1]){
+        level++;
       }
-      resolve(overallExp);
+      var levelExp = exp - expTable[level];
+      var totalExpNeeded = expTable[level + 1] - expTable[level];
+      var levelUpExp = totalExpNeeded - levelExp;
+      resolve([level, levelUpExp]);
     });
   });
 }
 
-function getLevelAndLevelUpExp(exp) {
-  /* Calculates current level and exp needed to next level.
-   * O(max_level) - computational complexity
-   */
-  var level = 0;
-  while(exp >= expTable[level + 1]){
-    level++;
-  }
-  var levelExp = exp - expTable[level];
-  var totalExpNeeded = expTable[level + 1] - expTable[level];
-  var levelUpExp = totalExpNeeded - levelExp;
-  return [level, levelUpExp];
-}
-
 function addSkill () {
   /* Adds skill to storage and to current table.
```

Comment:
What is current table? I guess you mean adding it to the view. If I'm right then this function should be placed in Views.js file instead of this one.